### PR TITLE
FCCEUD-81: Explain using attributes in Pv2

### PR DESCRIPTION
**Pull request by theashiot**

### CODE REVIEW

**Comment by theashiot** [comment]
@redhataccess/eud-tech-review Please review for technical completeness and accuracy.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+[role="_abstract"]
+You can define the document attributes in {ProductShortname} in an AsciiDoc file. You can then use these attributes in your content. When {ProductShortname} transforms the content to HTML, the attribute definitions replace the attributes. For example,
+if you define an attribute `:ProductName: Apollo` in your attributes file, and then use the attribute in your content as `\{ProductName} was released in 1969`, your content will render as `Apollo was released in 1969` in HTML.
+
+In {ProductShortname} you specify the path to your attribute file in the `pantheon2.yml` configuration file as follows:
+
+[source,yml,options="nowrap",subs="+quotes,attributes+"]
+----
+server: {server_url}
+# Example
```
Comment (noopurmath3) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:13]: `repository:` is required in pantheon2.yml file

**Comment by theashiot** [comment]
Hi @noopurmath3, thanks, i've updated the code block. Is it correct? Also is the documentation (commented out) for Jupiter correct? And should the target branch be main or master?

thanks,
ashwin

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+----
+repository: __<repository-name>__
+variants:
+   - name: __<product-name>__
+     attributes:
+       -  __<path-to-attribute-file-1>__.adoc 
+       -  __<path-to-attribute-file-2>__.adoc
+       # Examples
+       # _artifacts/document-attributes-common.adoc
+       # _artifacts/document-attributes-product.adoc
```
Comment (noopurmath3) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:48]: `variants` also has a `build=<true/false> in Jupiter to tell the system if we want to process this variant or not
I didnt add this earlier as this is just a part of the build.yml file, but adding `build` completes the attributes section for now

**Review by noopurmath3 — COMMENTED** [review]
Added a comment to add `build` in Jupiter build.yml file
Other than that this looks good to me
and the target branch should be `master`

**Comment by theashiot** [comment]
Thanks @noopurmath3, updated. Also retargeted to master. Have a look?

best,
ashwin

**Review by noopurmath3 — DISMISSED** [review]
lgtm 👍

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
+[id="con-about-document-attributes_{context}"]
+= About document attributes
+
+[role="_abstract"]
+You can define the document attributes in {ProductShortname} in an AsciiDoc file. You can then use these attributes in your content. When {ProductShortname} transforms the content to HTML, the attribute definitions replace the attributes. For example,
```
Comment (dvagnero) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:5]: I suggest adding one more step in this flow. You define the attributes in an adoc file, you include the attribute file in the pv2/jupiter config file, and then you can use these attributes. 

I suggest we use the same terminology as the asciidoc documentation, such as `:name-of-an-attribute: value of the attribute` https://docs.asciidoctor.org/asciidoc/latest/attributes/attribute-entries/ 
the value replaces/substitutes the attribute name

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     attributes:
+       -  __<path-to-attribute-file-1>__.adoc 
+       -  __<path-to-attribute-file-2>__.adoc
+       # Examples
+       # _artifacts/document-attributes-common.adoc
+       # _artifacts/document-attributes-product.adoc
+     build: true
+---- 
+
+When you set the path to your attributes file in in the `build.yml` configuration file, the attributes are available to each module and assembly in your repository. You don't need to explicitly include the attribute file in your module or assembly. If an attribute is defined in multiple files, {JupiterProductShortname} applies the definition in the  last file listed in the `build.yml` configuration file.
```
Comment (dvagnero) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:52]:
```suggestion
When you set the path to your attribute file in the `build.yml` configuration file, the attributes are available to each module and assembly in your repository. You don't need to explicitly include the attribute file in your module or assembly. If an attribute name is defined in multiple attribute files, {JupiterProductShortname} applies the definition in the last file listed in the `build.yml` configuration file. 
```

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+repository:__<repository_name>__
+    
+variants:
+   - name: global-attributes
+     path: __<path_to_attribute_file>__/__<attribute_file_name>__.adoc
+     # Example
+     # path: _artifacts/document-attributes.adoc
+---- 
+
+When you set the path to your attributes file in in the `pantheon2.yml` configuration file, the attributes are available to each module and assembly in your repository. You don't need to explicitly include the attribute file in your module or assembly.
```
Comment (dvagnero) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:24]: Global comment: is it attribute file or attributes file?

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 6]
```diff
@@ -0,0 +1,57 @@
+[id="con-about-document-attributes_{context}"]
+= About document attributes
```
Comment (dvagnero) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:2]: I suggest modifying the title. I'd say this module explains how to set the attribute file for your docs that are published in PVB2/Jupiter. The title "About document attributes" sounds like I'll learn about using attributes

**Review by dvagnero — CHANGES_REQUESTED** [review]
Hi @theashiot, great work. I added few comments and suggestions but it's all very minor stuff. Let me know if you have any questions.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc (hunk 5 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:24]: Globally changed to "attribute file"

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc (hunk 3 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:5]: excellent suggestions, updated.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc (hunk 4 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:52]: corrected

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc (hunk 6 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:2]: How about "About using document attributes in Pantheon"?

**Comment by theashiot** [comment]
Thanks a lot, @dvagnero for the awesome review, I have made the changes. Have a look?

best,

ashwin

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc (hunk 6 above)
Comment (dvagnero) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc]: Yes, I think that would be a nice fit

**Review by dvagnero — APPROVED** [review]
Good work @theashiot 
The changes look good to me. Thanks :)

**Comment by theashiot** [comment]
Thanks @dvagnero :D 

https://github.com/orgs/redhataccess/teams/pantheon-developers can you please take a look and merge?

thanks,

ashwin

**Review by stoobie — CHANGES_REQUESTED** [review]
@theashiot Looks pretty good. I didn't get through the entire Jupiter part.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 7]
```diff
@@ -0,0 +1,57 @@
+[id="con-about-document-attributes_{context}"]
+= About using document attributes in Pantheon
```
[the remark this answers is not in the document]
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc]: Done, i guess this is supported in Jupiter - I don't see anything against using attributes in headings unlike Pv2

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+When you set the path to your attribute file in the `pantheon2.yml` configuration file, the attributes are available to each module and assembly in your repository. You don't need to explicitly include the attribute file in your module or assembly. 
+
+// The below content is for Jupiter/Ganymede
+
+////
+= About using document attributes in Jupiter
+
+:JupiterProductShortname: Jupiter/Ganymede
+
+You can define the document attributes in {JupiterProductShortname} in an AsciiDoc file. You can then use these attributes in your content by including the attribute file in the {ProductShortname} config file. When {JupiterProductShortname} transforms the content to HTML, the attribute definitions replace the attributes. For example,
```
[the remark this answers is not in the document]
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc]: Done!

**Comment by theashiot** [comment]
Thanks, for the review @stoobie! I've made the required changes.

cheers,

ashwin

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc (hunk 2 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con-about-document-attributes.adoc:46]: Hi @stoobie, yep i added build:true in the coed block

**Pull request merged**